src/HttpHeader.cc

```diff
@@ -444,6 +452,7 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
                     char *p = (char *) this_line;   /* XXX Warning! This destroys original header content and violates specifications somewhat */
 
                     while ((p = (char *)memchr(p, '\r', field_end - p)) != NULL) {
+                        hasBareCr = "bare-CR";
```

Comment:
The "suspicious CR characters" condition has been established by the first memchr() call above already. There is no need to (re)set hasBareCr in a loop, for every such character.

The following suggestion looks huge, but all it does is moving the hasBareCr assignment line _before_ the loop-related code.

```suggestion
                    hasBareCr = "bare-CR";
                    char *p = (char *) this_line;   /* XXX Warning! This destroys original header content and violates specifications somewhat */

                    while ((p = (char *)memchr(p, '\r', field_end - p)) != NULL) {
```

Reply:
Fixed in 697651bd5